### PR TITLE
Insurgents - Change static chances to settings

### PR DESCRIPTION
**Pull request by Mike-MF**
- Changes all static chances to configurable settings.

Supersedes #526

### CODE REVIEW

**Comment by Mike-MF** [comment]
No idea what you mean by that one, but i also need to wrap this in a `waitUntilAndExecute` because editor units won't be affected by setting changes.

**Comment by jonpas** [comment]
Config entries in insurgent classes for chances. Next to gear definitions.

**Comment by Mike-MF** [comment]
This is configurable, which really helps since a lot of the time i use these as just generic factions with an assortment of gear. Config entries would just remain static.

**Comment by jonpas** [comment]
That's why I ask, I am fine with either.

File: addons/insurgents/XEH_preInit.sqf [hunk 1]
```diff
@@ -37,4 +37,10 @@ private _configFile = configFile >> QGVAR(gear);
     } forEach ("true" configClasses _x);
 } forEach [_configFile, missionConfigFile >> QGVAR(gear)];
 
+// Class EH for randomised gear
+["CBA_settingsInitialized", {
+    [QGVAR(soldier_i), "init", FUNC(randomizeGear), true, [], true] call CBA_fnc_addClassEventHandler;
+    diag_log "[TAC]: Event Fired";
```
Comment (jonpas) [inline addons/insurgents/XEH_preInit.sqf]:
```suggestion
```

Comment (veteran29) [inline addons/insurgents/XEH_preInit.sqf]: Might want to get rid of that debug line.

**Review by jonpas — APPROVED** [review]

**Pull request merged**